lib/internal/Magento/Framework/DB/ExpressionConverter.php

```diff
@@ -103,40 +105,45 @@ public static function addTranslate($from, $to)
      */
     public static function shortenEntityName($entityName, $prefix)
     {
-        if ($entityName !== null && strlen($entityName) > self::MYSQL_IDENTIFIER_LEN) {
-            $shortName = ExpressionConverter::shortName($entityName);
-            if (strlen($shortName) > self::MYSQL_IDENTIFIER_LEN) {
-                // md5() here is not for cryptographic use.
-                // phpcs:ignore Magento2.Security.InsecureFunction
-                $hash = md5($entityName);
-                if (strlen($prefix . $hash) > self::MYSQL_IDENTIFIER_LEN) {
-                    $entityName = self::trimHash($hash, $prefix, self::MYSQL_IDENTIFIER_LEN);
-                } else {
-                    $entityName = $prefix . $hash;
-                }
-            } else {
-                $entityName = $shortName;
-            }
+        if ($entityName === null) {
+            return null;
+        }
+
+        $fullEntityName = $prefix . $entityName;
+        if (strlen($fullEntityName) <= self::MYSQL_IDENTIFIER_LEN) {
+            return $fullEntityName;
+        }
+
+        $shortName = ExpressionConverter::shortName($fullEntityName);
+        if (strlen($shortName) <= self::MYSQL_IDENTIFIER_LEN) {
+            return $shortName;
         }
-        return $entityName;
+
+        // md5() here is not for cryptographic use.
+        // phpcs:ignore Magento2.Security.InsecureFunction
+        $hash = md5($entityName);
+        $hashedName = $prefix . $hash;
+        if (strlen($hashedName) <= self::MYSQL_IDENTIFIER_LEN) {
+            return $hashedName;
+        }
+
+        $trimmedHash = self::trimHash($hash, $prefix, self::MYSQL_IDENTIFIER_LEN);
```

Comment:
`trimHash` method call is provided 3 parameters, but the original method signature uses 1 parameter.

Reply:
Also suggesting you introduce a new method instead of making changes in the existing method `trimHash`, as removing existing arguments from the method might be backward incompatible. 

Also, mark `trimHash` method deprecate.

Reply:
This is a `private` method (`trimHash`) so it does not need to be marked as deprecated.